pkg/obfuscate/appsec.go

```diff
@@ -0,0 +1,538 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package obfuscate
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// ObfuscateAppSec obfuscates the given appsec tag value in order to remove
+// sensitive values from the appsec security events.
+func (o *Obfuscator) ObfuscateAppSec(val string) string {
+	keyRE := o.opts.AppSec.ParameterKeyRegexp
+	valueRE := o.opts.AppSec.ParameterValueRegexp
+	if keyRE == nil && valueRE == nil {
+		return val
+	}
+	ob := appsecEventsObfuscator{
+		keyRE:   keyRE,
+		valueRE: valueRE,
+	}
+	output, err := ob.obfuscate(val)
+	if err != nil {
+		o.log.Errorf("unexpected error while obfuscating the appsec events: %v", err)
+		return val
+	}
+	return output
+}
+
+type appsecEventsObfuscator struct {
+	keyRE, valueRE *regexp.Regexp
+}
+
+type unexpectedScannerOpError int
+
+func (err unexpectedScannerOpError) Error() string {
+	return fmt.Sprintf("unexpected json scanner operation %d", err)
+}
+
+// inputDiff holds the list of diffs, built by the obfuscator, and to ultimately
```

Comment:
Can we just call it `diff`? Or `edits`?

Also, it'd be nice to be a bit more "stupid" about its definition. Since "diff" can bear a lot of complexity in a readers mind (thinking of the UNIX diff, or git diff). Perhaps defining it as:

```go
// diff specifies a set of changes consisting of byte offsets and a string to replace them in another input.
```

Reply:
Fixed in https://github.com/DataDog/datadog-agent/commit/11c457b0780a7cc343f26f42329eb8bda61771da